spec/system/admin/reports/sales_tax/sales_tax_totals_by_order_spec.rb

```diff
@@ -0,0 +1,395 @@
+# frozen_string_literal: true
+
+require 'system_helper'
+
+describe "Sales Tax Totals By order" do
+  #  Scenarion 1: added tax
+  #  1 producer
+  #  1 distributor
+  #  product that costs 100$
+  #  shipping costs 10$
+  #  the packaging cost is 5$
+  #  1 order with 1 line item
+  #  the line item match 2 tax rates: country (2.5%) and state (1.5%)
+  let!(:table_header){
+    [
+      "Distributor",
+      "Order Cycle",
+      "Order ID",
+      "Tax Category",
+      "Tax Rate Name",
+      "Tax Rate",
+      "Total excl. Tax ($)",
+      "Tax",
+      "Total incl. Tax ($)"
+    ].join(" ").upcase
+  }
+  let!(:state_zone){ create(:zone_with_state_member) }
+  let!(:country_zone){ create(:zone_with_member) }
+  let!(:tax_category){ create(:tax_category) }
+  let!(:state_tax_rate){ create(:tax_rate, zone: state_zone, tax_category: tax_category) }
+  let!(:country_tax_rate){ create(:tax_rate, zone: country_zone, tax_category: tax_category) }
+  let!(:ship_address){ create(:ship_address) }
+
+  let!(:variant){ create(:variant) }
+  let!(:product){ variant.product }
+  let!(:supplier){ create(:supplier_enterprise) }
+  let!(:distributor){ create(:distributor_enterprise_with_tax) }
+  let!(:distributor_fee){ create(:enterprise_fee, :flat_rate, amount: 5) }
+  let!(:payment_method){ create(:payment_method, :flat_rate) }
+  let!(:shipping_method){ create(:shipping_method, :flat_rate, amount: 10) }
+
+  let!(:order){ create(:order_with_distributor, distributor: distributor) }
+  let!(:order_cycle){
+    create(:simple_order_cycle, suppliers: [supplier], distributors: [distributor],
+                                variants: [variant])
+  }
+
+  let(:admin){ create(:admin_user) }
+
+  before do
```

Comment:
Is it ok to disable `Metrics/BlockLength: Block has too many lines` here?

Reply:
I would first look to see if the block can be reduced in size. I've not reviewed the code (will await the dependent PR), but just quickly scanning there are a lot of `update!` statements. Can these details be incorporated into the `let!`s above? 

Eg instead of
```
customer1.update!({ first_name: 'cfname', last_name: 'clname', code: 'ABC123' })
```

Can it be moved to:
```
  let!(:customer1){ create(:customer, enterprise: create(:enterprise), user: create(:user), first_name: 'cfname', last_name: 'clname', code: 'ABC123' ) }
```
(sorry if the syntax is wrong, I'm not very experienced with FactoryBot yet)
(also, this could be reformatted over multiple lines for readability).

There's also a lot of comments, but I encourage you to keep them because I think it's great to explain what the code is doing! 👍

Reply:
@dacook 
line 100=> 109 can be removed if those [changes](https://github.com/openfoodfoundation/openfoodnetwork/pull/10304#discussion_r1086628329) are accepted.

The linter in this case will no longer complain.